.github/workflows/pull-code-checks.yml

```diff
@@ -0,0 +1,39 @@
+name: PR Code checks
+
+on:
+  pull_request:
+    branches:
+      - "main"
+      - "release-*"
+    paths-ignore:
+      - 'docs/**'
+      - '**.md'
+  workflow_dispatch:
+
+jobs:
+  unit-tests:
+
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+
+    - name: Prepare Test
+      uses: "./.github/template/prepare-checks"
+
+    - name: run tests
+      run: make test
+
+  linting:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-go@v4
+        with:
+          go-version: '1.21'
+          cache: false
+      - name: golangci-lint
+        uses: golangci/golangci-lint-action@v3
+        with:
+          install-mode: binary
+          version: latest
+          args: --timeout=5m --config=./.golangci.yaml 
```

Comment:
You added a new make target to download the linter binary, but here we are using an action with preinstalled linter. Is that target only for local usage?

Reply:
Initially I used the binary which was not present in GH action, so I added it.
It turned out that the linting rules changed in latest versions, so I prefer to use the binary for local execution in latest version always to have same results as the github action